### PR TITLE
DOC: Document hints, kind, and labels

### LINKED ISSUE #614: add documentation: hints, kind, and labels
As part of the ALS Hackathon, add documentation for the ``hints``, ``kind``, and ``labels`` attributes.  

TODO

* [x] add ``hints`` subsection under Device to device-overview.rst
* [x] add links to ``hints`` from other sections in ophyd and bluesky
* [x] add ``kind`` section to signals.rst (but that doc is very limited!)
* [ ] add links to ``kind`` from other sections in ophyd and bluesky
* [x] in ophyd.device.Device, ``Kind`` needs a link to its class
* [x] add ``labels`` section to signals.rst (but that doc is very limited!)
* [ ] add links to ``labels`` from other sections in ophyd and bluesky

Take into account that ``labels`` might be deprecated in the near future.

### PR DESCRIPTION
**Pull request by prjemian**
fixes #614 - should squash merge

### CODE REVIEW

**Comment by danielballan** [comment]
Thanks, @prjemian! Will review in detail next week but wanted to quickly say thanks a bunch for this.

File: doc/source/device-overview.rst [hunk 1]
```diff
@@ -171,11 +173,139 @@ and static information about the object
...
+that advise how that device should be handled by the callback.
+While it could be used for many purposes, its first use has been to
+direct the selection of the relevant axes and signals to use when plotting
+data from an event stream.
+
+There are two different locations where the 
+``hints`` dictionary is created.
+
+1. during the specification of an ophyd :class:`~device.Device`
+1. configuration of the ``start`` document by a :mod:`bluesky` plan
```
Comment (klauer) [inline doc/source/device-overview.rst]:
```suggestion
2. Configuration of the ``start`` document by a :mod:`bluesky` plan
```

File: doc/source/device-overview.rst [hunk 2]
```diff
@@ -171,11 +173,139 @@ and static information about the object
...
+provides information about an :mod:`ophyd` device to :mod:`bluesky` callbacks
+that advise how that device should be handled by the callback.
+While it could be used for many purposes, its first use has been to
+direct the selection of the relevant axes and signals to use when plotting
+data from an event stream.
+
+There are two different locations where the 
+``hints`` dictionary is created.
+
+1. during the specification of an ophyd :class:`~device.Device`
```
Comment (klauer) [inline doc/source/device-overview.rst]:
```suggestion
1. During the specification of an ophyd :class:`~device.Device`
```

File: doc/source/device-overview.rst [hunk 3]
```diff
@@ -171,11 +173,139 @@ and static information about the object
...
+is ``"rectilinear"``, then use LiveGrid, otherwise use LivePlot.
+
 
-:class:`Component`
-------------------
+:class:`~device.Component`
+--------------------------------
 
-The :class:`Compent` class is a python descriptor_ which override the
+The :class:`~device.Component` class is a python descriptor_ which override the
```
Comment (klauer) [inline doc/source/device-overview.rst]:
```suggestion
The :class:`~device.Component` class is a python descriptor_ which overrides the
```

File: doc/source/device-overview.rst [hunk 4]
```diff
@@ -171,11 +173,139 @@ and static information about the object
...
 behavior on attribute access.  This allows us to use a declarative
```
Comment (klauer) [inline doc/source/device-overview.rst]:
```suggestion
behavior of attribute access.  This allows us to use a declarative
```

File: ophyd/epics_motor.py [hunk 5]
```diff
@@ -284,6 +284,6 @@ def report(self):
 class MotorBundle(Device):
     """Sub-class this to device a bundle of motors
 
-    This provides better default behavior for ``hints``.
+    This provides better default behavior for :ref:`hints``.
```
Comment (klauer) [inline ophyd/epics_motor.py]:
```suggestion
    This provides better default behavior for :ref:``hints``.
```

**Pull request merged**